docs: fix naming - use API Client instead of SDK (#1175)
hyperledger/cactus: CONTRIBUTING.md

> Be aware that glob patterns need quoting in some operating system's shell environments and not necessarily on others.
> In the npm scripts Cactus uses we quote all of them to ensure a wider shell compatibility.

> ### Building the API Client

Reviewer: @Leeyoungone 

```suggestion
### Building the API Client(s)
```

(because when we originally wrote this documentation there was going to be only one API client but then we ended up creating many)

Author: Thank you for making suggestions for the documentation 😭 This might have been the hardest PR I've picked up because I just don't know what I was suppose to write.

You do not need to do anything special to have the API Client sources generated and
compiled. It is all part of the `npm run build:dev:backend` task which you can run yourself
or as part of the CI script (`./tools/ci.sh`).

> The API Client is itself just another package named `openapi.json` and can be dependend on by

Reviewer: @Leeyoungone 

```suggestion
The API client code is automatically generated from the respective `openapi.json` file of each package that exposes any web services (REST/SocketIO/gRPC/etc.) and can be depended on by
```

> other packages where applicable. There's a dedicated package that is meant to contain common functionality among the API Clients. 

Reviewer: @Leeyoungone 

```suggestion
other packages where applicable. There's a dedicated `@hyperledger/cactus-api-client` package that is meant to contain common functionality among the rest of API clients. The concept here is similar to abstract classes and their sub-class implementations.
```

> Each `openapi.json` provices its own API Client via the code generator. 

Reviewer: @Leeyoungone 

```suggestion
Each `openapi.json` produces its own API client via the code generator that also contains relevant model definitions such as interfaces describing the request/response bodies of all possible operations and validation constraints as well.
```

> The API Client is designed to be a universal package meaning that it runs just fine in

Reviewer: @Leeyoungone 

```suggestion
The API clients are designed to be a universal components meaning that it runs just fine in
```

browser and also NodeJS environments. This is very important as we do not wish

> to maintain two (or more) separate API Client codebases and we also want as much of it

Reviewer: @Leeyoungone 

```suggestion
to maintain two (or more) separate API client codebases for the various platforms and we also want as much of it
```
